cpp/modmesh/buffer/SimpleArray.cpp

```diff
@@ -478,6 +509,46 @@ SimpleArrayPlex & SimpleArrayPlex::operator=(SimpleArrayPlex && other) noexcept
     return *this;
 }
 
+size_t SimpleArrayPlex::alignment() const
+{
+    if (m_instance_ptr == nullptr)
+    {
+        return 0;
+    }
+
+    switch (m_data_type)
+    {
+    case DataType::Bool:
```

Comment:
A question just occurred to me: when we want to add struct array support, this switch case will be a problem.

But we worry about it when needing to support struct array.  We do not worry about it now.

Reply:
It's not critical atm.  Discuss on dc if you want: https://discord.com/channels/730297880140578906/730393342818648204/1442850883401416705